src/typescript-service.ts

```diff
@@ -1022,26 +1022,14 @@ export class TypeScriptService {
 						if (entry.sortText) {
 							item.sortText = entry.sortText;
 						}
-						const details = configuration.getService().getCompletionEntryDetails(fileName, offset, entry.name);
-						if (details) {
-							item.documentation = ts.displayPartsToString(details.documentation);
-							item.detail = ts.displayPartsToString(details.displayParts);
-							if (this.supportsCompletionWithSnippets) {
-								item.insertTextFormat = InsertTextFormat.Snippet;
-								if (entry.kind === 'property') {
-									item.insertText = details.name;
-								} else {
-									const parameters = details.displayParts
-										.filter(p => p.kind === 'parameterName')
-										.map((p, i) => '${' + `${i + 1}:${p.text}` + '}');
-									const paramString = parameters.join(', ');
-									item.insertText = details.name + `(${paramString})`;
-								}
-							} else {
-								item.insertTextFormat = InsertTextFormat.PlainText;
-								item.insertText = details.name;
-							}
-						}
+
+						// context for future resolve requests:
+						item.data = {
+							uri,
+							offset,
+							entryName: entry.name
+						};
```

Comment:
could you define an interface for this / add this to a custom CompletionItem interface in request-type?